integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -228,9 +230,12 @@ class CarbonScanRDD[T: ClassTag](
       statistic.addStatistics(QueryStatisticsConstants.BLOCK_ALLOCATION, System.currentTimeMillis)
       statisticRecorder.recordStatisticsForDriver(statistic, queryId)
       statistic = new QueryStatistic()
-      val carbonDistribution = CarbonProperties.getInstance().getProperty(
+      var carbonDistribution = CarbonProperties.getInstance().getProperty(
         CarbonCommonConstants.CARBON_TASK_DISTRIBUTION,
         CarbonCommonConstants.CARBON_TASK_DISTRIBUTION_DEFAULT)
+      if (directScan) {
```

Comment:
I think there is too many path, if directScan can always save memory and copy less, I suggest we always use it, then no need for adding one more configuration